repos/system_upgrade/el7toel8/actors/vimcheck/actor.py

```diff
@@ -0,0 +1,37 @@
+from leapp.actors import Actor
+from leapp.libraries.common.reporting import report_generic
+from leapp.models import InstalledRedHatSignedRPM, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class VimCheck(Actor):
+    """
+    Check if vim-minimal or vim-enhanced is installed and if so, inform about planned actions.
+    """
+
+    name = 'vim_check'
+    consumes = (InstalledRedHatSignedRPM,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        vim_pkgs = ['vim-minimal', 'vim-enhanced']
+        is_any_vim_pkg_installed = False
+
+        # FIXME: use has_pkg from repo shared library once it is implemented
+        rh_rpms = [pkg.name for pkg in next(self.consume(InstalledRedHatSignedRPM), InstalledRedHatSignedRPM()).items]
+
+        for pkg in vim_pkgs:
+            if pkg in rh_rpms:
+                is_any_vim_pkg_installed = True
+
+        if is_any_vim_pkg_installed:
```

Comment:
Better yet - a one-liner replacing lines 19,24-28 (though with new `has_package` functionality):
```suggestion
        if any(has_package(InstalledRedHatSignedRPM, pkg) for pkg in vim_pkgs):
```